src/OrchardCore.Modules/OrchardCore.Shortcodes/Services/TemplateShortcodeProvider.cs

```diff
@@ -44,7 +50,21 @@ public async ValueTask<string> EvaluateAsync(string identifier, Arguments argume
             return await _liquidTemplateManager.RenderAsync(template.Content, _htmlEncoder, model,
                 scope =>
                 {
-                    scope.SetValue("Content", model.Content);
+                    // Used for recursion checking.
+                    scope.SetValue(identifier, "");
+
+                    // Don't conflict with the liquid scope 'Content' property.
+                    var content = scope.GetValue("Content").ToObjectValue();
+                    if (content is LiquidContentAccessor contentAccessor)
```

Comment:
Looks good @jtkech 

But. Need to check now, how will this work with a nested shortcode? 
Is it a local scope it is attaching too? I think so, but checking.

e.g.
```
[foo]foocontent[bar]barcontent[/bar][/foo]
```

Reply:
@deanmarcussen it works as each get and use its own local scope